app/components/primer/beta/button.rb

```diff
@@ -68,7 +68,10 @@ class Button < Primer::Component
       renders_one :trailing_visual, types: {
         icon: Primer::Beta::Octicon,
         label: Primer::Beta::Label,
-        counter: Primer::Beta::Counter
+        counter: lambda { |**system_arguments|
+          @trailing_visual_counter = true
+          Primer::Beta::Counter.new("aria-hidden": true, **system_arguments)
```

Comment:
I am directly setting `aria-hidden` on the counter, because when I nested this inside a span, it caused issues. See https://github.com/primer/view_components/pull/2807#discussion_r1581274450